source/git/git_package_release_guide.rst

```diff
@@ -0,0 +1,58 @@
+Releasing a Package on Github
+=============================
+
+This guide details the pre release checks to undertake before releasing a package on github, and the actual release process on github. The guide requires no prior knowledge of creating software releases on Github. While this guide is developed for Github, the ideas themselves can be applied to releases on other git management systems (such as Gitlab).
+
+Preliminaries
+~~~~~~~~~~~~~
+
+The release version of the package is created by creating a stable self-contained version of the package (called a *Release*), tagging it with a version and creating accompanying metadata such as Release Notes and Changelogs. Depending on the development environment, Changelogs and version tags can be automated. Release notes are targeted at the end user, highlighting the features of the new release, whereas a Changelog is usually a detailed description of all changes since the last release version.
+
+The standard convention for naming versions (also known as `Semantic Versioning <https://semver.org/>`__) is to tag a release as MAJOR.MINOR.PATCH where
+
+1. MAJOR refers to an update to the whole package that is backwards incompatible with the previous versions.
+2. MINOR refers to either new features, functional updates, or a large number of bundled bug fixes.
+3. PATCH refers to small bug fixes, hotfixes, or minor changes in the code. 
+
+On github, creating a release usually entails creating a stable version of the package on a branch, then creating a release tag for the specific stable commit. These branches can be standard main/develop or a dedicated release branch, which is usually dependent on the conventions of the organization creating the package. 
+
+Pre-Release Checks
+~~~~~~~~~~~~~~~~~~
+
+Prior to release, consider the following checks on your github repo:
+
+1. Is the package on a stable release branch? Or will the package release be on the main/develop branch of the repo?
+2. Does the release branch have any conflicts with the main/develop branches?
+3. Do all the tests pass?
+4. Does the release version have any dependencies/links to local files that are not in the repo?
```

Comment:
I think another thing to add is that the installation instructions are detailed, with the pre-requisites specificed in the pyproject.toml file.